src/base_object-inl.h

```diff
@@ -250,6 +250,17 @@ BaseObjectPtrImpl<T, kIsWeak>& BaseObjectPtrImpl<T, kIsWeak>::operator=(
   return *new (this) BaseObjectPtrImpl(std::move(other));
 }
 
+template <typename T, bool kIsWeak>
+BaseObjectPtrImpl<T, kIsWeak>::BaseObjectPtrImpl(std::nullptr_t)
+    : BaseObjectPtrImpl() {}
+
+template <typename T, bool kIsWeak>
+BaseObjectPtrImpl<T, kIsWeak>& BaseObjectPtrImpl<T, kIsWeak>::operator=(
+    std::nullptr_t) {
+  this->~BaseObjectPtrImpl();
+  return *this;
```

Comment:
I think this is technically UB? I'd maybe just make this 

```suggestion
  return *this = BaseObjectPtrImpl();
```

so that it's more obviously correct

Reply:
Ah, right.. good catch. It looked a bit funny on review.

Reply:
Thanks, fixed!